File: src/main/java/it/infordata/certs/domain/Certs.java
package it.infordata.certs.domain;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

import javax.persistence.*;

import java.io.Serializable;
import java.time.LocalDate;

/**
 * A Certs.
 */
@Entity
@Table(name = "certs")
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
public class Certs implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sequenceGenerator")
    @SequenceGenerator(name = "sequenceGenerator")
    private Long id;

    @Column(name = "title")
    private String title;

    @Column(name = "content")
    private String content;

    @Column(name = "main_content")
    private String mainContent;

    @Lob
    @Column(name = "attachments")
    private byte[] attachments;

    @Column(name = "attachments_content_type")
    private String attachmentsContentType;

    @Lob
    @Column(name = "images")
    private byte[] images;

    @Column(name = "images_content_type")
    private String imagesContentType;

    @Column(name = "date")
    private LocalDate date;

    // jhipster-needle-entity-add-field - JHipster will add fields here
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public Certs title(String title) {
        this.title = title;
        return this;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public Certs content(String content) {
        this.content = content;
        return this;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getMainContent() {
        return mainContent;
    }

    public Certs mainContent(String mainContent) {
        this.mainContent = mainContent;
        return this;
    }

    public void setMainContent(String mainContent) {
        this.mainContent = mainContent;
    }

    public byte[] getAttachments() {
        return attachments;
    }

    public Certs attachments(byte[] attachments) {
        this.attachments = attachments;
        return this;
    }

    public void setAttachments(byte[] attachments) {
        this.attachments = attachments;
    }

    public String getAttachmentsContentType() {
        return attachmentsContentType;
    }

    public Certs attachmentsContentType(String attachmentsContentType) {
        this.attachmentsContentType = attachmentsContentType;
        return this;
    }

    public void setAttachmentsContentType(String attachmentsContentType) {
        this.attachmentsContentType = attachmentsContentType;
    }

    public byte[] getImages() {
        return images;
    }

    public Certs images(byte[] images) {
        this.images = images;
        return this;
    }

    public void setImages(byte[] images) {
        this.images = images;
    }

    public String getImagesContentType() {
        return imagesContentType;
    }

    public Certs imagesContentType(String imagesContentType) {
        this.imagesContentType = imagesContentType;
        return this;
    }

    public void setImagesContentType(String imagesContentType) {
        this.imagesContentType = imagesContentType;
    }

    public LocalDate getDate() {
        return date;
    }

    public Certs date(LocalDate date) {
        this.date = date;
        return this;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }
    // jhipster-needle-entity-add-getters-setters - JHipster will add getters and setters here

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof Certs)) {
            return false;
        }
        return id != null && id.equals(((Certs) o).id);
    }

    @Override
    public int hashCode() {
        return 31;
    }

    // prettier-ignore
    @Override
    public String toString() {
        return "Certs{" +
            "id=" + getId() +
            ", title='" + getTitle() + "'" +
            ", content='" + getContent() + "'" +
            ", mainContent='" + getMainContent() + "'" +
            ", attachments='" + getAttachments() + "'" +
            ", attachmentsContentType='" + getAttachmentsContentType() + "'" +
            ", images='" + getImages() + "'" +
            ", imagesContentType='" + getImagesContentType() + "'" +
            ", date='" + getDate() + "'" +
            "}";
    }
}

File: ui/widgets/certs/formWidget/src/components/__types__/certs.js
import PropTypes from 'prop-types';

export default PropTypes.shape({
  id: PropTypes.number,

  title: PropTypes.string,
  content: PropTypes.string,
  mainContent: PropTypes.string,
  attachments: PropTypes.string,
  images: PropTypes.string,
  date: PropTypes.string,
});

export const formValues = PropTypes.shape({
  title: PropTypes.string,
  content: PropTypes.string,
  mainContent: PropTypes.string,
  attachments: PropTypes.string,
  images: PropTypes.string,
  date: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(Date)]),
});

export const formTouched = PropTypes.shape({
  title: PropTypes.oneOfType([PropTypes.bool, PropTypes.shape()]),
  content: PropTypes.oneOfType([PropTypes.bool, PropTypes.shape()]),
  mainContent: PropTypes.oneOfType([PropTypes.bool, PropTypes.shape()]),
  attachments: PropTypes.oneOfType([PropTypes.bool, PropTypes.shape()]),
  images: PropTypes.oneOfType([PropTypes.bool, PropTypes.shape()]),
  date: PropTypes.oneOfType([PropTypes.bool, PropTypes.shape()]),
});

export const formErrors = PropTypes.shape({
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.shape()]),
  content: PropTypes.oneOfType([PropTypes.string, PropTypes.shape()]),
  mainContent: PropTypes.oneOfType([PropTypes.string, PropTypes.shape()]),
  attachments: PropTypes.oneOfType([PropTypes.string, PropTypes.shape()]),
  images: PropTypes.oneOfType([PropTypes.string, PropTypes.shape()]),
  date: PropTypes.oneOfType([PropTypes.string, PropTypes.shape()]),
});
